# **Stepik "Поколение Python": курс для начинающих**

# ***Ввод-вывод данных***

## **Целочисленная арифметика. Часть 1**

### Три последовательных числа
<p>Напишите программу вывода на экран трех последовательно идущих чисел, каждое на отдельной строке. Первое число вводит пользователь, остальные числа вы должны сами вычислять в программе.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести три последовательно идущих числа в соответствии с условием задачи.</p>

In [2]:
x = input("")
print(x)
print(int(x)+1)
print(int(x)+2)

1
1
2
3


### Сумма трёх чисел
<p>Напишите программу, которая считывает три целых числа и выводит на экран их сумму. Каждое число записано в отдельной строке.</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести сумму введённых чисел.</p>

In [ ]:
a = input()
b = input()
c = input()
print(int(a)+int(b)+int(c))

1
2
3
6


### Куб 🧊
<p>Напишите программу, вычисляющую объём куба и площадь его полной поверхности по введенному значению длины ребра.</p>

<p>Формат входных данных<br>
На вход программе подается одно целое число – длина ребра.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст и числа в соответствии с условием задачи.</p>

<p>Примечание 1. Объем куба и площадь полной поверхности можно вычислить по формулам:<br>
V = a^3<br>

S = 6 * a^2</p>

<p>Примечание 2. Обратите внимание на то, что на текущем этапе обучения мы не знаем про оператор возведения в степень,
поэтому пользуемся определением степени числа – число умножается само на себя указанное количество раз. Например:<br>

𝑎^4 = a * a * a * a</p>

In [ ]:
a = int(input())
print("Объем = ", a*a*a)
print("Площадь полной поверхности = ", 6*a*a)

25
Объем =  15625
Площадь полной поверхности =  3750


### Значение функции
<p>Напишите программу, которая по введённым целым значениям 𝑎 и 𝑏 вычисляет значение следующего выражения:<br>

3(𝑎+𝑏)^3 + 275𝑏^2 − 127𝑎 − 41</p>

<p>Формат входных данных<br>
На вход программе подаются два целых числа (каждое на отдельной строке):
𝑎 и 𝑏.</p>

<p>Формат выходных данных<br>
Программа должна вывести значение функции по введённым числам 𝑎 и 𝑏.</p>

<p>Примечание. Обратите внимание на то, что на текущем этапе обучения мы не знаем про оператор возведения в степень,
поэтому пользуемся определением степени числа – число умножается само на себя указанное количество раз. Например:<br>

𝑎^4 = a * a * a * a</p>

In [ ]:
a = int(input())
b = int(input())
x = 3 * ((a + b) * (a + b) * (a + b)) + 275 * (b * b) - 127 * a - 41
print(x)

1
0
-165


### Следующее и предыдущее 🔢
<p>Напишите программу, которая считывает целое число и выводит для него на экран следующее и предыдущее целые числа в следующем формате:<br>

Следующее за числом <текущее число> число: <следующее число><br>
Для числа <текущее число> предыдущее число: <предыдущее число></p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [ ]:
a = int(input())
print("Следующее за числом", a, "число:", a + 1)
print(f"Для числа {a} предыдущее число: {a - 1}")

1
Следующее за числом 1 число: 2
Для числа 1 предыдущее число: 0


### Стоимость покупки 🛒
<p>Напишите программу, которая считает стоимость трёх компьютеров, состоящих из монитора, системного блока, клавиатуры и мыши.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых числа (каждое на отдельной строке):
стоимость монитора, стоимость системного блока, стоимость клавиатуры, стоимость мыши.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – стоимость покупки (трёх компьютеров).</p>

In [ ]:
mon = int(input())
sys = int(input())
key = int(input())
mouse = int(input())
print(3 * (mon + sys + key + mouse))

9900
55600
3999
2990
217467


### Арифметические операции
<p>Напишите программу, которая принимает на вход два числа 𝑎 и 𝑏, вычисляет сумму, разность и произведение для этих чисел и выводит текст в следующем формате:<br>

<число a> + <число b> = <сумма чисел a и b><br>
<число a> - <число b> = <разность чисел a и b><br>
<число a> * <число b> = <произведение чисел a и b></p>

<p>Формат входных данных<br>
На вход программе подаются два целых числа (каждое на отдельной строке):
𝑎 и 𝑏.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [ ]:
a = int(input())
b = int(input())
print(f"{a} + {b} = {a + b}")
print(f"{a} - {b} = {a - b}")
print(f"{a} * {b} = {a * b}")

2
7
2 + 7 = 9
2 - 7 = -5
2 * 7 = 14


### Арифметическая прогрессия
<p>Арифметической прогрессией называется последовательность чисел a1, a2, ..., an, каждое из которых, начиная с 𝑎2, получается из предыдущего путем прибавления к нему одного и того же постоянного числа 𝑑 (разность прогрессии).
Если известен первый член прогрессии (𝑎1) и её разность (𝑑), то 𝑛-ый член арифметической прогрессии находится по формуле:<br>
an = a1 + d(n − 1)</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа:<br>
𝑎1, 𝑑 и 𝑛, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести 𝑛-ый член арифметической прогрессии.</p>

<p>Примечание. Арифметической прогрессией является, например, следующая последовательность:<br>
−6,−3,0,3,6,9,12</p>

In [ ]:
a1 = int(input())
d = int(input())
n = int(input())
print(a1 + d * (n - 1))

-1
1
2
0


### Разделяй и властвуй
<p>Напишите программу, которая считывает целое положительное число 𝑥 и выводит на экран последовательность чисел x,2x,3x,4x и 5𝑥, разделённых тремя черточками.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое положительное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст согласно условию задачи.</p>

In [ ]:
a = int(input())
print(a, 2*a, 3*a, 4*a, 5 * a, sep = "---")

7
7---14---21---28---35


## **Целочисленная арифметика. Часть 2**

In [ ]:
"""Геометрическая прогрессия
Геометрической прогрессией называется последовательность чисел 𝑏1, 𝑏2, …, 𝑏𝑛, каждое из которых, начиная с 𝑏2,
получается из предыдущего умножением на одно и то же постоянное число 𝑞 (знаменатель прогрессии).

Если известен первый член прогрессии и ее знаменатель, то 𝑛-ый член геометрической прогрессии находится по формуле
𝑏𝑛 = 𝑏1 ⋅ 𝑞^(𝑛−1)

Входные данные
На вход программе подаются три целых числа:
𝑏1, 𝑞 и 𝑛 , каждое на отдельной строке.

Выходные данные
Программа должна вывести 𝑛-ый член геометрической прогрессии.
"""
b1 = int(input())
q = int(input())
n = int(input())
print(b1 * q ** (n - 1))

1
2
5
16


In [ ]:
"""Расстояние в метрах
Напишите программу, которая находит полное число метров по заданному числу сантиметров.

Формат входных данных
На вход программе подаётся натуральное число – количество сантиметров.

Формат выходных данных
Программа должна вывести одно число – полное число метров.
"""
b = int(input())
print(b // 100)

300
3


In [ ]:
"""Мандарины 🍊
𝑛 школьников делят k мандаринов поровну, неделящийся остаток остается в корзине.
Сколько целых мандаринов достанется каждому школьнику?
Сколько целых мандаринов останется в корзине?

Формат входных данных
На вход программе подаётся два целых числа: количество школьников и количество мандаринов, каждое на отдельной строке.

Формат выходных данных
Программа должна вывести два числа: количество мандаринов, которое достанется каждому школьнику, и количество мандаринов, которое останется в корзине, каждое на отдельной строке.
"""
a = int(input())
b = int(input())
print(b // a)
print(b % a)

12
6
0
6


In [ ]:
"""Сама неотвратимость
Безумный титан Танос собрал все 6 камней бесконечности и намеревается уничтожить половину населения Вселенной по щелчку пальцев.
При этом если население Вселенной является нечетным числом, то титан проявит милосердие и округлит количество выживших в большую сторону.
Помогите Мстителям подсчитать количество выживших.

Формат входных данных
На вход дается число целое
n – население Вселенной.

Формат выходных данных
Программа должна вывести одно число – количество выживших.
"""
a = int(input())
print(a - (a // 2))

99
50


In [ ]:
"""Пересчёт временного интервала ⌚
Напишите программу для пересчёта величины временного интервала, заданного в минутах, в величину, выраженную в часах и минутах в следующем формате:

<исходное кол-во минут> мин - это <полное кол-во часов> час <оставшееся кол-во минут> минут.
Формат входных данных
На вход программе подаётся целое число – количество минут.

Формат выходных данных
Программа должна вывести текст в соответствии с условием задачи.
"""
a = int(input())
print(f"{a} мин - это {a // 60} час {a % 60} минут.")

50
50 мин - это 0 час 50 минут.


In [ ]:
"""Номер купе 🚉🌶️
В купейном вагоне имеется 9 купе с четырьмя местами для пассажиров в каждом.
Напишите программу, которая определяет номер купе, в котором находится место с заданным номером
(нумерация мест сквозная, начинается с 1).

Формат входных данных
На вход программе подаётся целое число – место с заданным номером в вагоне.

Формат выходных данных
Программа должна вывести одно число – номер купе, в котором находится указанное место.
"""
a = int(input())
print((a + 3) // 4)
# print(-((-a) // 4))

9
3


In [ ]:
"""Трёхзначное число
Напишите программу, которая рассчитывает сумму и произведение цифр положительного трёхзначного числа и выводит текст в следующем формате:

Сумма цифр = <сумма цифр>
Произведение цифр = <произведение цифр>
Формат входных данных
На вход программе подаётся положительное трёхзначное число.

Формат выходных данных
Программа должна вывести текст в соответствии с условием задачи.
"""
a = int(input())
last = a % 10
mid = (a // 10) % 10
fir = a // 100
print("Сумма цифр =", last + mid + fir)
print("Произведение цифр =", last * mid * fir)

333
Сумма цифр = 9
Произведение цифр = 27


In [ ]:
"""Перестановка цифр
Дано трехзначное число abc, в котором все цифры различны.
Напишите программу, которая выводит шесть чисел, образованных при перестановке цифр заданного числа.

Формат входных данных
На вход программе подаётся положительное трёхзначное целое число, все цифры которого различны.

Формат выходных данных
Программа должна вывести шесть чисел, образованных при перестановке цифр заданного числа, в следующем порядке (каждое на новой строке):
𝑎𝑏𝑐, 𝑎𝑐𝑏, 𝑏𝑎𝑐, 𝑏𝑐𝑎, 𝑐𝑎𝑏, 𝑐𝑏𝑎
"""
num = int(input())
c = num % 10
b = (num // 10) % 10
a = num // 100
print(a, b, c, sep = '')
print(a, c, b, sep = '')
print(b, a, c, sep = '')
print(b, c, a, sep = '')
print(c, a, b, sep = '')
print(c, b, a, sep = '')

123
123
132
213
231
312
321


In [ ]:
"""Четырёхзначное число
Напишите программу для нахождения цифр четырёхзначного числа.

Формат входных данных
На вход программе подаётся положительное четырёхзначное целое число.

Формат выходных данных
Программа должна вывести текст в соответствии с условием задачи.
"""
a = int(input())
last = a % 10
mid2 = (a // 10) % 10
mid1 = (a // 100) % 10
fir = a // 1000
print("Цифра в позиции тысяч равна", fir)
print("Цифра в позиции сотен равна", mid1)
print("Цифра в позиции десятков равна", mid2)
print("Цифра в позиции единиц равна", last)

1234
Цифра в позиции тысяч равна 1
Цифра в позиции сотен равна 2
Цифра в позиции десятков равна 3
Цифра в позиции единиц равна 4


# ***Итоговая работа на ввод-вывод данных***

### Прямоугольник
<p>Напишите программу, которая выводит прямоугольник, по периметру состоящий из звёздочек (*).</p>

<p>Примечание. Высота и ширина прямоугольника равны 4 и 17 звёздочкам соответственно.</p>
<p>Тестовые данные 🟢</p>
<p>Sample Input:<br>
</p>

<p>Sample Output:<br>

```
*****************
*               *
*               *
*****************
```

</p>

In [ ]:
print("*****************")
print("*               *")
print("*               *")
print("*****************")

*****************
*               *
*               *
*****************


### Квадрат суммы 🆚 Сумма квадратов
<p>Напишите программу, которая считывает два целых числа a и b и выводит на экран квадрат суммы (a+b)^2 и сумму квадратов (𝑎^2) + (b^2) этих чисел в следующем формате:</p>

<p>Напишите программу, которая считывает четыре целых положительных числа a,b,c и d и выводит на экран значение выражения<br>
(𝑎^𝑏) + (𝑐^𝑑).</p>

<p>Квадрат суммы a и b равен <квадрат суммы a и b><br>
Сумма квадратов a и b равна <сумма квадратов a и b></p>
<p>Формат входных данных<br>
На вход программе подаются два целых числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием.</p>

In [ ]:
a, b = int(input()), int(input())
print(f"Квадрат суммы {a} и {b} равен {(a + b) ** 2}")
print(f"Сумма квадратов {a} и {b} равна {a ** 2 + b ** 2}")

3
2
Квадрат суммы 3 и 2 равен 25
Сумма квадратов 3 и 2 равна 13


### Большое число
<p>Как известно, целые числа в языке Python не имеют ограничений, которые встречаются в других языках программирования.</p>

<p>Напишите программу, которая считывает четыре целых положительных числа a,b,c и d и выводит на экран значение выражения<br>
(𝑎^𝑏) + (𝑐^𝑑).</p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых положительных числа a,b,c и 𝑑, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести значение (𝑎^𝑏) + (𝑐^𝑑).</p>

In [ ]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
print(a ** b + c ** d)

9
29
7
27
4710194409608608369201743232


### Размножение n-ок
<p>Напишите программу, которая считывает целое положительное число 𝑛, 𝑛 ∈ [1;9] и выводит значение числа n + nn + nnn.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое положительное число 𝑛, 𝑛 ∈ [1;9].</p>

<p>Формат выходных данных<br>
Программа должна вывести число n + nn + nnn.</p>

<p>Примечание. Для первого теста<br>
1+11+111=123.</p>

In [ ]:
a = input()
b = a + a
c = a + a + a
print(int(a) + int(b) + int(c))

1
123


# ***Условный оператор***

## ***Выбор из двух***

### Пароль 🔒
<p>При регистрации на сайтах требуется вводить пароль дважды. Это сделано для безопасности, поскольку такой подход уменьшает возможность неверного ввода пароля.<br>

Напишите программу, которая сравнивает пароль и его подтверждение. Если они совпадают, то программа выводит: «Пароль принят», иначе: «Пароль не принят».</p>

<p>Формат входных данных<br>
На вход программе подаются две строки.</p>

<p>Формат выходных данных<br>
Программа должна вывести одну строку в соответствии с условием задачи.</p>

In [3]:
a, b = input(), input()
if a == b:
    print("Пароль принят")
else:
    print("Пароль не принят")


qwerty
qwerty
Пароль принят


### Четное или нечетное?
<p>Напишите программу, которая определяет, является число четным или нечетным.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести «Четное» (без кавычек), если число четное, или «Нечетное» (без кавычек) в противном случае.</p>

In [5]:
a = int(input())
if a % 2 == 0:
    print("Четное")
else:
    print("Нечетное")

5
Нечетное


### Соотношение
<p>Напишите программу, которая проверяет, что для заданного четырехзначного числа выполняется следующее соотношение: сумма первой и последней цифр равна разности второй и третьей цифр.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое положительное четырёхзначное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести «ДА», если соотношение выполняется, и «НЕТ» — если не выполняется.</p>

In [9]:
a = int(input())
last = a % 10
mid2 = (a // 10) % 10
mid1 = (a // 100) % 10
fir = a // 1000
if fir + last == mid1 - mid2:
    print("ДА")
else:
    print("НЕТ")

1614
ДА


### Роскомнадзор 🔞
<p>Напишите программу, которая определяет, разрешён ли пользователю доступ к интернет-ресурсу или нет.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое число – возраст пользователя.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст «Доступ разрешен», если возраст пользователя не менее 18 лет, или «Доступ запрещен» в противном случае.</p>

In [11]:
a = int(input())
if a < 18:
    print("Доступ запрещен")
else:
    print("Доступ разрешен")

11
Доступ запрещен


### Арифметическая прогрессия
<p>Напишите программу, которая определяет, являются ли три заданных числа (в указанном порядке) последовательными членами арифметической прогрессии.</p>

<p>Формат входных данных<br>
На вход программе подаются три числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести «YES» или «NO» (без кавычек) в соответствии с условием задачи.</p>

In [14]:
a, b, c = int(input()), int(input()), int(input())
if (a + c) / 2 * 3 != a + b + c:
    print("NO")
else:
    print("YES")

2
4
7
NO


### Наименьшее из двух чисел
<p>Напишите программу, которая определяет наименьшее из двух чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются два различных целых числа.</p>

<p>Формат выходных данных<br>
Программа должна вывести наименьшее из двух чисел.</p>

In [15]:
a, b = int(input()), int(input())
if a < b:
    print(a)
else:
    print(b)

44
33
33


### Наименьшее из четырёх чисел 🌶️
<p>Напишите программу, которая определяет наименьшее из четырёх чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых числа.</p>

<p>Формат выходных данных<br>
Программа должна вывести наименьшее из четырёх чисел.</p>

<p>Примечание. Учитывайте, что минимальные числа могут дублироваться (смотрите тест №5). В таком случае необходимо вывести только одно из них.</p>

In [24]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
# print(min(a, b, c, d))

if a < b:
    b = a
else:
    a = b

if c < d:
    d = c
else:
    c = d

if a < d:
    d = a
else:
    a = d
print(a)

100
200
4
66
4


### Возрастная группа 👧🏻
<p>Напишите программу, которая по введённому возрасту пользователя сообщает, к какой возрастной группе он относится:<br>
до 13 (включительно) – детство;<br>
от 14 до 24 (включительно) – молодость;<br>
от 25 до 59 (включительно) – зрелость;<br>
от 60 (включительно) – старость.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число – возраст пользователя.</p>

<p>Формат выходных данных<br>
Программа должна вывести название возрастной группы.</p>

In [25]:
a = int(input())
if a <= 13:
    print("детство")
elif 13 < a <= 24:
    print("молодость")
elif 24 < a <= 59:
    print("зрелость")
else:
    print("старость")

# print("старость" if n>59 else "зрелость" if n>24 else "молодость" if n>13 else "детство")

14
молодость


### Только + 🌶️
<p>Напишите программу, которая считывает три числа и подсчитывает сумму только положительных чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа, каждое на новой строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – сумму положительных чисел.</p>

<p>Примечание. Если положительных чисел нет, то следует вывести 0.</p>

In [29]:
a, b, c = int(input()), int(input()), int(input())

if a <= 0:
    a = 0
if b <= 0:
    b = 0
if c <= 0:
    c = 0
print(a + b + c)

# print((a if a>0 else 0) + (b if b>0 else 0) + (c if c>0 else 0))

4
-22
1
5


## ***Логические операции***

### Принадлежность 1
<p>Напишите программу, которая принимает целое число x и определяет, принадлежит ли данное число указанному промежутку. </p>

<p>Формат входных данных<br>
На вход программе подаётся целое число x.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/9bd3a69a-d538-41a2-aefb-e76386c653f9/" style="height: 120px; width:400px;"/>

<p>Примечание. Если точка выколотая, то граница не включается. Если же точка закрашенная, то граница включается.</p>

In [3]:
a = int(input())
if a > -1 and a < 17:
    print("Принадлежит")
else:
    print("Не принадлежит")

Принадлежит


### Принадлежность 2
<p>Напишите программу, которая принимает целое число x и определяет, принадлежит ли данное число указанным промежуткам.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое число x.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/5d994476-13fb-40aa-a529-4fd12c9e424a/" style="height: 120px; width:400px;"/>

<p>Примечание. Если точка выколотая, то граница не включается. Если же точка закрашенная, то граница включается.</p>

In [6]:
a = int(input())
if a <= -3 or a >= 7:
    print("Принадлежит")
else:
    print("Не принадлежит")

Принадлежит


### Принадлежность 3
<p>Напишите программу, которая принимает целое число x и определяет, принадлежит ли данное число указанным промежуткам.</p>

<p>Формат входных данных<br>
На вход программе подаётся целое число x.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/4659a972-d8f5-4186-90ed-38dfdacf17fb/" style="height: 120px; width:400px;"/>

<p>Примечание. Если точка выколотая, то граница не включается. Если же точка закрашенная, то граница включается.</p>

In [10]:
a = int(input())
if -30 < a <= -2 or 7 < a <= 25:
    print("Принадлежит")
else:
    print("Не принадлежит")

Не принадлежит


### Красивое число 🌶️
<p>Назовем число красивым, если оно является четырехзначным и делится нацело на 7 или на 17. Напишите программу, определяющую, является ли введённое число красивым. Программа должна вывести «YES», если число является красивым, или «NO» в противном случае.</p>

<p>Формат входных данных<br>
На вход программе подаётся натуральное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [16]:
a = int(input())
if 1000 <= a < 10000 and (a % 7 == 0 or a % 17 == 0):
    print("YES")
else:
    print("NO")

NO


### Неравенство треугольника
<p>Напишите программу, которая принимает три положительных числа и определяет, существует ли невырожденный треугольник с такими сторонами.</p>

<p>Формат входных данных<br>
На вход программе подаётся три положительных целых числа.</p>

<p>Формат выходных данных<br>
Программа должна вывести «YES» или «NO» в соответствии с условием задачи.</p>

<p>Примечание. Треугольник существует, если выполняется неравенство треугольника:<br>
a+b>c<br>
a+c>b<br>
b+c>a<br>
где a,b,c - стороны треугольника.</p>

In [18]:
a, b, c = int(input()), int(input()), int(input())
if a + b > c and a + c > b and b + c > a:
    print("YES")
else:
    print("NO")

YES


### Високосный год
<p>Напишите программу, которая определяет, является ли год с данным номером високосным. Если год является високосным, то выведите «YES», иначе выведите «NO».<br>
Год является високосным, если его номер кратен 4, но не кратен 100, или если он кратен 400.</p>

<p>Формат входных данных<br>
На вход программе подаётся натуральное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [19]:
a = int(input())
if a % 400 == 0 or (a % 4 == 0 and not a % 100 == 0):
    print("YES")
else:
    print("NO")

YES


### Ход ладьи ♖
<p>Даны две различные клетки шахматной доски. Напишите программу, которая определяет, может ли ладья попасть с первой клетки на вторую одним ходом. Программа получает на вход четыре числа от 1 до 8 каждое, задающие номер столбца и номер строки сначала для первой клетки, потом для второй клетки. Программа должна вывести «YES», если из первой клетки ходом ладьи можно попасть во вторую, или «NO» в противном случае.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре числа от 1 до 8.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/c13b2aac-577c-45a5-b29e-7136842cb955/" style="height: 200px; width:200px;"/>

<p>Примечание. Шахматная ладья ходит по горизонтали и вертикали.</p>

In [21]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
if a == c or b == d:
    print("YES")
else:
    print("NO")

NO


### Ход короля ♔🌶️
<p>Даны две различные клетки шахматной доски. Напишите программу, которая определяет, может ли король попасть с первой клетки на вторую одним ходом. Программа получает на вход четыре числа от 1 до 8 каждое, задающие номер столбца и номер строки сначала для первой клетки, потом для второй клетки. Программа должна вывести «YES», если из первой клетки ходом короля можно попасть во вторую, или «NO» в противном случае.</p>

<p>Формат входных данных<br>
На вход программе подаются четыре числа от 1 до 8.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/e759d185-a6d3-47c8-96b2-7b52fa03af09/" style="height: 200px; width:200px;"/>

<p>Примечание. Шахматный король ходит по горизонтали, вертикали и диагонали, но только на 1 клетку.</p>

In [23]:
a, b, c, d = int(input()), int(input()), int(input()), int(input())
if -1 <= a - c <= 1 and -1 <= b - d <= 1:
    print("YES")
else:
    print("NO")

NO


## ***Вложенные и каскадные условия***

### Гонка спидстеров
<p>Зум бросил вызов Флэшу и предложил ему честный поединок в виде гонки вокруг магнетара. В случае проигрыша эта нейтронная звезда зарядится и уничтожит мир, поэтому Флэш решил не рисковать без причины и узнать у своего друга Циско Рамона, есть ли смысл принимать вызов. Циско получил данные, что скорость Зума равна n, а скорость Флэша равна k.<br>
Напишите программу, которая должна вывести ответ Циско на вопрос Флэша.</p>

<p>Формат входных данных<br>
На вход программе подаётся два целых числа n и k – скорости Зума и Флэша.</p>

<p>Формат выходных данных<br>
Если Зум быстрее Флэша, нужно вывести «NO», а если Флэш быстрее Зума, нужно вывести «YES». В случае же если их скорости равны, нужно вывести «Don't know».</p>

In [24]:
a, b = int(input()), int(input())
if a < b:
    print("YES")
elif a > b:
    print("NO")
else:
    print("Don't know")

NO


### Вид треугольника 📐
<p>Напишите программу, которая классифицирует треугольник на основе длин его сторон. Программа должна принимать три числа, каждое из которых представляет собой длину одной из его сторон. В результате программа должна определить, является ли треугольник равносторонним, равнобедренным или разносторонним.</p>

<p>Формат входных данных<br>
На вход программе подаются три числа (каждое на отдельной строке) – длины сторон существующего треугольника.</p>

<p>Формат выходных данных<br>
Программа должна вывести на экран текст – тип треугольника («Равносторонний», «Равнобедренный» или «Разносторонний»).</p>

<img src="https://ucarecdn.com/c2e79215-c75e-46d7-bf94-d08bd37e12aa/" style="height: 200px; width:400px;"/>

In [25]:
a, b, c = int(input()), int(input()), int(input())
if a == b == c:
    print("Равносторонний")
elif a == b or a == c or c == b:
    print("Равнобедренный")
else:
    print("Разносторонний")

Равнобедренный


### Серединное число
<p>Даны три различных целых числа. Напишите программу, которая находит серединное значение из этих чисел.</p>

<p>Формат входных данных<br>
На вход программе подаются три различных целых числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести серединное значение из набора трёх чисел.</p>

<p>Примечание. Серединное значение набора чисел — число, которое находится в середине этого набора, если его упорядочить по возрастанию.</p>

In [27]:
a, b, c = int(input()), int(input()), int(input())
if a >= b >= c or c >= b >= a:
    print(b)
elif b >= a >= c or c >= a >= b:
    print(a)
else:
    print(c)

20


### Количество дней 📅
<p>Дан порядковый номер месяца (1,2,…, 12). Напишите программу, которая выводит на экран количество дней в этом месяце. Принять, что год является невисокосным.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число – порядковый номер месяца.</p>

<p>Формат выходных данных<br>
Программа должна вывести количество дней в этом месяце.</p>

<p>Примечание. Постарайтесь написать программу так, чтобы в ней было не более трёх условий.</p>

In [32]:
a = int(input())
if a == 2:
    print(28)
elif a == 4 or a == 6 or a == 9 or a == 11:
    print(30)
else:
    print(31)

30


### Церемония взвешивания
<p>Известен вес боксера-любителя (целое число). Известно, что вес таков, что боксер может быть отнесён к одной из трех весовых категорий:<br>
Легкий вес – до 60 кг (невключительно);<br>
Первый полусредний вес – до 64 кг (невключительно);<br>
Полусредний вес – до 69 кг (невключительно).<br>
Напишите программу, определяющую, в какой категории будет выступать данный боксер.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст – название весовой категории.</p>

In [33]:
a = int(input())
if a < 60:
    print("Легкий вес")
elif a < 64:
    print("Первый полусредний вес")
elif a < 69:
    print("Полусредний вес")

Первый полусредний вес


### Самописный калькулятор 🌶️
<p>Напишите программу, которая считывает с клавиатуры два целых числа и строку. Если эта строка является обозначением одной из четырёх математических операций (+, -, *, /), то выведите результат применения этой операции к введённым ранее числам, в противном случае выведите «Неверная операция» (без кавычек). Если пользователь захочет поделить на ноль, выведите текст «На ноль делить нельзя!» (без кавычек).</p>

<p>Формат входных данных<br>
На вход программе подаются два целых числа и строка, все на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести результат применения операции к введенным числам или соответствующий текст, если операция неверная либо если происходит деление на ноль.</p>

In [35]:
a, b, c = int(input()), int(input()), input()
if c == '+':
    print(a + b)
elif c == '-':
    print(a - b)
elif c == '*':
    print(a * b)
elif c == '/' and b == 0:
    print("На ноль делить нельзя!")
elif c == '/':
    print(a / b)
else:
    print("Неверная операция")

6.6


### Цветовой микшер 🌶️
<p>Красный, синий и желтый называются основными цветами, потому что их нельзя получить путем смешения других цветов. При смешивании двух основных цветов получается вторичный цвет:<br>
если смешать красный и синий, то получится фиолетовый;<br>
если смешать красный и желтый, то получится оранжевый;<br>
если смешать синий и желтый, то получится зеленый.<br>
Напишите программу, которая считывает названия двух основных цветов для смешивания. Если пользователь вводит что-нибудь помимо названий «красный», «синий» или «желтый», то программа должна вывести сообщение об ошибке. В противном случае программа должна вывести название вторичного цвета, который получится в результате.</p>

<p>Формат входных данных<br>
На вход программе подаются две строки, каждая на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести полученный цвет смешения либо сообщение «ошибка цвета», если введён был не цвет.</p>

<p>Примечание 1. Если смешать красный и красный, то получится красный и т.д.</p>

<p>Примечание 2. Поиграйтесь с настоящим цветовым микшером 🙂</p>

In [36]:
a, b= input(), input()
if (a == 'красный' and b == 'красный') or (b == 'синий' and a == 'синий') or (a == 'желтый' and b == 'желтый'):
    print(a)
elif (a == 'красный' and b == 'синий') or (b == 'красный' and a == 'синий'):
    print("фиолетовый")
elif (a == 'красный' and b == 'желтый') or (b == 'красный' and a == 'желтый'):
    print("оранжевый")
elif (a == 'желтый' and b == 'синий') or (b == 'желтый' and a == 'синий'):
    print("зеленый")
else:
    print("ошибка цвета")

ошибка цвета


### Цвета колеса рулетки 🌶️
<p>На колесе рулетки карманы пронумерованы от 0 до 36. Ниже приведены цвета карманов: <br>
карман 0 зеленый;<br>
для карманов с 1 по 10 карманы с нечетным номером имеют красный цвет, карманы с четным номером – черный;<br>
для карманов с 11 по 18 карманы с нечетным номером имеют черный цвет, карманы с четным номером – красный;<br>
для карманов с 19 по 28 карманы с нечетным номером имеют красный цвет, карманы с четным номером – черный;<br>
для карманов с 29 по 36 карманы с нечетным номером имеют черный цвет, карманы с четным номером – красный.<br>
Напишите программу, которая считывает номер кармана и показывает, является ли этот карман зеленым, красным или черным. Программа должна вывести сообщение об ошибке, если пользователь вводит число, которое лежит вне диапазона от 0 до 36.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно целое число.</p>

<p>Формат выходных данных<br>
Программа должна вывести цвет кармана либо сообщение «ошибка ввода», если введённое число лежит вне диапазона от 0 до 36.</p>

<img src="https://ucarecdn.com/d53ddcd9-bebd-4413-a6f8-0a4cbc64963a/" style="height: 200px;"/>

In [38]:
a = int(input())
if a == 0:
    print("зеленый")
elif (1 <= a <= 10 or 19 <= a <= 28) and a % 2 != 0:
    print("красный")
elif (1 <= a <= 10 or 19 <= a <= 28) and a % 2 == 0:
    print("черный")
elif (11 <= a <= 18 or 29 <= a <= 36) and a % 2 != 0:
    print("черный")
elif (11 <= a <= 18 or 29 <= a <= 36) and a % 2 == 0:
    print("красный")
else:
    print("ошибка ввода")

красный


### Пересечение отрезков 🌶️🌶️
<p>На числовой прямой даны два отрезка: [a1; b1] и [a2; b2]. Напишите программу, которая находит их пересечение.<br>
Пересечением двух отрезков может быть:<br>
отрезок: <br>
<img src="https://ucarecdn.com/67067971-5281-478c-9d14-a6f05715da42/" style="height: 200px;"/><br>
точка: <br>
<img src="https://ucarecdn.com/ac19322f-64ea-4877-b064-295623d83179/" style="height: 200px;"/><br>
пустое множество: <br>
<img src="https://ucarecdn.com/04e13a22-3eb0-48bc-bf89-bf5cbd1ab332/" style="height: 200px;"/></p>

<p>Формат входных данных<br>
На вход программе подаются четыре целых числа a1, b1, a2, b2, каждое на отдельной строке. Гарантируется, что 𝑎1 < 𝑏1 и 𝑎2 < 𝑏2.</p>

<p>Формат выходных данных<br>
Программа должна вывести на экран границы отрезка, являющегося пересечением, либо общую точку, либо текст «пустое множество».</p>

In [49]:
a1, b1, a2, b2 = int(input()), int(input()), int(input()), int(input())
if a2 == b1:
    print(a2)
elif a1 == b2:
    print(a1)
elif a1 <= a2 < b1 <= b2:
    print(a2, b1)
elif a2 <= a1 < b2 <= b1:
    print(a1, b2)
elif a1 <= a2 and b2 <= b1:
    print(a2, b2)
elif a2 <= a1 and b1 <= b2:
    print(a1, b1)
else:
    print("пустое множество")

22 25


# ***Итоговая работа на условный оператор***

# ***Типы данных***

## ***Числовые типы данных: int, float***

### Площадь треугольника
<p>Напишите программу, которая считывает длины двух катетов в прямоугольном треугольнике и выводит его площадь.</p>

<p>Формат входных данных<br>
На вход программе подаются два действительных числа (каждое на отдельной строке) – длины катетов.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – площадь треугольника.</p>

<img src="https://ucarecdn.com/c2fcd9b1-7105-4a8f-af91-0ede0f9f5b2e/" style="height: 200px; width:400px;"/>

In [52]:
a, b = float(input()), float(input())
print(0.5 * a * b)

344988.0


### Две старушки 👵
<p>Две старушки идут навстречу друг другу с постоянными скоростями 𝑉1 и V2 км/ч. Определите, через какое время (в часах) старушки встретятся, если расстояние между ними равно 𝑆 км.</p>

<p>Формат входных данных<br>
На вход программе подаются три действительных числа 𝑆, 𝑉1, 𝑉2, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число в соответствии с условием задачи.</p>

<p>Примечание. Это очень быстрые старушки.</p>

<img src="https://ucarecdn.com/ac06613a-0c35-4020-9116-323b77f37494/" style="height: 200px; width:400px;"/>

In [2]:
s, a, b = float(input()), float(input()), float(input())
print(s / (a + b))

2.0


### Обратное число
<p>Напишите программу, которая считывает с клавиатуры одно число и выводит обратное ему. Если при этом введённое с клавиатуры число – ноль, то вывести «Обратного числа не существует» (без кавычек).</p>

<p>Формат входных данных<br>
На вход программе подаётся одно действительное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести действительное число обратное данному, либо текст в соответствии с условием задачи.</p>

<img src="https://ucarecdn.com/4df00cbb-37df-4955-9f01-4da08fd44cad/" style="height: 200px; width:400px;"/>

In [3]:
a = float(input())
if a == 0:
    print("Обратного числа не существует")
else:
    print(1 / a)

0.4


### 451 градус по Фаренгейту 🌡️
<p>У известного американского писателя Рэя Бредбери есть роман «451 градус по Фаренгейту». Напишите программу, которая определяет, какой температуре по шкале Цельсия соответствует указанное значение по шкале Фаренгейта.<br>
Для получения градусов по Цельсию из градусов по Фаренгейту используйте следующую формулу:<br>
𝑡𝐶=5/9(𝑡𝐹−32)<br>
где 𝑡𝐹 – температура в градусах по Фаренгейту.</p>

<p>Формат входных данных<br>
На вход программе подаётся одно действительное число – температура в градусах по шкале Фаренгейта.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно действительное число – температуру в градусах по шкале Цельсия.</p>

In [4]:
a = float(input())
print(5/9 * (a - 32))

232.5277777777778


### Dog age 🐶
<p>На вход программе подаётся число n – количество собачьих лет. Напишите программу, которая вычисляет возраст собаки в человеческих годах по следующему алгоритму: в течение первых двух лет собачий год равен 10.5 человеческим годам, после этого каждый год собаки равен 4 человеческим годам.</p>

<p>Формат входных данных<br>
На вход программе подаётся натуральное число – количество собачьих лет.</p>

<p>Формат выходных данных<br>
Программа должна вывести возраст собаки в человеческих годах.</p>

In [7]:
a = float(input())
if (a <= 2):
    print(a * 10.5)
else:
    print(21 + ((a - 2) * 4))

25.0


### Первая цифра после точки
<p>Дано положительное действительное число. Выведите его первую цифру после десятичной точки.</p>

<p>Формат входных данных<br>
На вход программе подаётся положительное действительное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести цифру в соответствии с условием задачи.</p>

In [11]:
a = float(input())
print(int(a * 10) % 10)

2


### Дробная часть
<p>Дано положительное действительное число. Выведите его дробную часть.</p>

<p>Формат входных данных<br>
На вход программе подаётся положительное действительное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести дробную часть числа в соответствии с условием задачи.</p>

<p>Примечание. Числа с плавающей точкой (тип float) в Python хранятся в неточном виде, что связано с ограниченным количеством выделенных для них байт в памяти. Поэтому ваша программа может выдавать дробную часть от числа не в ожидаемом виде. Например, для числа 44.45 остаток может выглядеть как:<br>
0.45000000000000284<br>
Наша проверяющая система засчитает как 0.45, так и 0.45000000000000284 правильными ответами.</p>

In [13]:
a = float(input())
print(a - int(a))

0.19999999999708962


### Наибольшее и наименьшее
<p>Напишите программу, которая находит наименьшее и наибольшее из пяти чисел.</p>

<p>Формат входных данных<br>
На вход программе подается пять целых чисел, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести наименьшее и наибольшее число с поясняющей надписью.</p>

In [16]:
a, b, c, d, e = int(input()), int(input()), int(input()), int(input()), int(input())
print("Наименьшее число = ", min(a, b, c, d, e), "\n", "Наибольшее число = ", max(a, b, c, d, e), sep="")

Наименьшее число = 1
Наибольшее число = 5


### Сортировка трёх 🌶️
<p>Напишите программу, которая упорядочивает три числа от большего к меньшему.</p>

<p>Формат входных данных<br>
На вход программе подаются три целых числа, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести три числа, каждое на отдельной строке, упорядоченных от большего к меньшему.</p>

<p>Примечание. Учитывайте, что числа могут быть равны.</p>

In [18]:
a, b, c = int(input()), int(input()), int(input())
if min(a, b, c) * 3 == a + b + c or max(a, b, c) * 3 == a + b + c:
    print(a, b, c, sep="\n")
elif min(a, b, c) * 2 + max(a, b, c) == a + b + c:
    print(max(a, b, c), min(a, b, c), min(a, b, c), sep="\n")
elif min(a, b, c) + max(a, b, c) * 2 == a + b + c:
    print(max(a, b, c), max(a, b, c), min(a, b, c), sep="\n")
else:
    print(max(a, b, c))
    if a != max(a, b, c) and a != min(a, b, c):
        print(a)
    elif b != max(a, b, c) and b != min(a, b, c):
        print(b)
    else:
        print(c)
    print(min(a, b, c))

# a, b, c = int(input()), int(input()), int(input())
# min, max = min(a, b, c), max(a, b, c)
# print(max, a + b + c - min - max, min, sep = '\n')

160
150
150


### Интересное число
<p>Назовём число интересным, если в нём разность максимальной и минимальной цифры равняется средней по величине цифре. Напишите программу, которая определяет, интересное число или нет. Если число интересное, следует вывести «Число интересное», иначе – «Число неинтересное».</p>

<p>Формат входных данных<br>
На вход программе подается целое трёхзначное число.</p>

<p>Формат выходных данных<br>
Программа должна вывести текст в соответствии с условием задачи.</p>

In [21]:
a = int(input())
a1 = a % 10
a2 = (a // 10) % 10
a3 = a // 100
if max(a1, a2, a3) - min(a1, a2, a3) == (a1 + a2 + a3) - min(a1, a2, a3) - max(a1, a2, a3):
    print("Число интересное")
else:
    print("Число неинтересное")

Число неинтересное


### Абсолютная сумма
<p>Даны пять чисел 𝑎1, 𝑎2, 𝑎3, 𝑎4, 𝑎5. Напишите программу, которая вычисляет сумму их модулей ∣a1∣ + ∣a2∣ + ∣a3∣ + ∣a4∣ + ∣a5∣.</p>

<p>Формат входных данных<br>
На вход программе подаются пять действительных чисел 𝑎1, 𝑎2, 𝑎3, 𝑎4, 𝑎5, каждое на отдельной строке.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – сумму модулей введённых чисел.</p>

In [ ]:
a, b, c, d, e = float(input()), float(input()), float(input()), float(input()), float(input())
print(abs(a) + abs(b) + abs(c) + abs(d) + abs(e))

### Манхэттенское расстояние
<p>Прогуливаясь по Манхэттену, вы не можете попасть из точки А в точку Б по кратчайшему пути. Если только вы не умеете проходить сквозь стены, вам обязательно придется идти вдоль его параллельно-перпендикулярных улиц.<br>
На плоскости манхэттенское расстояние между двумя точками (p1;p2) и (q1;q2) определяется так: ∣p1 − q1∣ + ∣p2 − q2∣.<br>
Напишите программу, определяющую манхэттенское расстояние между двумя точками, координаты которых заданы.</p>

<p>Формат входных данных<br>
На вход программе подается четыре целых числа, каждое на отдельной строке – p1, p2, q1, q2.</p>

<p>Формат выходных данных<br>
Программа должна вывести одно число – манхэттенское расстояние.</p>

<img src="https://ucarecdn.com/d3720269-108b-48fb-9ca5-f24af59fb358/" style="height: 200px;"/>

In [22]:
p1, p2, q1, q2 = int(input()), int(input()), int(input()), int(input())
print(abs(p1 - q1) + abs(p2 - q2))

13


## ***Строковый тип данных***